controllers/collect_vision_data/collect_vision_data.cpp

```diff
@@ -0,0 +1,317 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2021 NUbots <nubots@nubots.net>
+ */
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <array>
+#include <filesystem>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <random>
+#include <sstream>
+#include <webots/Camera.hpp>
+#include <webots/Motor.hpp>
+#include <webots/PositionSensor.hpp>
+#include <webots/Robot.hpp>
+#include <webots/Supervisor.hpp>
+
+#include "yaml-cpp/yaml.h"
+
+std::string padLeft(int number, int width) {
+    std::stringstream ss;
+    ss << std::setw(width) << std::setfill('0') << number;
+    return ss.str();
+}
+
+const int QUALITY = 100;
```

Comment:
Should probably comment this

Reply:
Also, is this a 100 quality jpeg? That's crazy talk!
My guess is for the actual competition you'll be down at like 75 or less

Reply:
Its also used for PNGs I think

Reply:
PNGs are just always 100% quality, they're a lossless compression

Reply:
Just had a look at the docs. They ignore the quality parameter for PNGs (as they should).